src/components/__tests__/ResultView.test.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { describe, it, expect } from 'vitest';
+import { render, screen } from '@testing-library/react';
+import { ResultView } from '../ResultView';
+import { SearchResult } from '../../types';
+
+describe('ResultView', () => {
+	const createResult = (
+		overrides: Partial<SearchResult> = {}
+	): SearchResult => ({
+		pageTitle: '',
+		targetText: '',
+		revisionId: null,
+		loading: false,
+		error: null,
+		language: 'en',
+		...overrides,
+	});
+
+	it('displays loading state', () => {
+		const loadingResult = createResult({ loading: true });
+		render(<ResultView result={loadingResult} />);
+
+		expect(screen.getByText(/searching/i)).toBeTruthy();
```

Comment:
TODO: consider to get by role

ref. https://testing-library.com/docs/queries/byrole